8.0 Add sale_rental_lot_selection + several fixes and improvements in sale_rental

alexis-via opened the pull request.

alexis-via committed:
Add module sale_rental_lot_selection

Adapt sale_rental and sale_order_lot_selection to make them work together from a functional point of view

alexis-via committed:
Fix auto-configuration of multiple warehouses for rental

alexis-via committed:
Warehouse setup for rental v2

alexis-via committed:
Restore initial code in sale_order_lot_selection/model/sale.py

alexis-via committed:
Add OCA as author

Warning -> UserError
Fix changes in sale_order_lot_section

alexis-via committed:
Adapt tests of sale_order_lot_selection

alexis-via committed:
sale_rental: Add store=True on some fields, in particular 'state'

Add search view for rental
FIX compatibility between sale_rental and sale_order_lot_selection

alexis-via commented:
@eLBati I made the work required to have travis green with my module sale_rental_lot_selection. At last !

alexis-via committed:
Add rental state "cancel"

Better string

sebastienbeau committed:
[FIX] fix perf issue with sale_order_lot_selection

alexis-via committed:
Make it work with older version of Odoo 8

eLBati commented:
@alexis-via hi, github says "This branch has conflicts that must be resolved". Could you check?

alexis-via commented:
mmm, the conflict is in product_id_change_with_wh... so it has to be studied in depth and I need to re-test everything :-(

jabibi committed:
[FIX] populate the 'Return Stock Move' field; module version

When creating the rental record, 'Return Stock Move' field was not
taking the corresponding reference to the incoming stock.move

Also fixed the version number and added Odoo Community Association (OCA)
to author list

jabibi commented on sale_order_lot_selection/model/sale.py:
```diff
@@ -45,24 +55,28 @@ def product_id_change_with_wh(
+        res.update({'domain': {'lot_id': [
+            ('id', 'in', available_lots),
+            ('product_id', '=', product.id),
+            ]}})
```
I tested with this piece of code and seems to work :ok_hand: 
Please look at the setdefault part in order to maintain previous values

```
    @api.multi
    def product_id_change_with_wh(
            self, pricelist, product_id, qty=0,
            uom=False, qty_uos=0, uos=False, name='', partner_id=False,
            lang=False, update_tax=True, date_order=False, packaging=False,
            fiscal_position=False, flag=False, warehouse_id=False):
        res = super(SaleOrderLine, self).product_id_change_with_wh(
            pricelist, product_id, qty=qty, uom=uom,
            qty_uos=qty_uos, uos=uos, name=name, partner_id=partner_id,
            lang=lang, update_tax=update_tax,
            date_order=date_order, packaging=packaging,
            fiscal_position=fiscal_position, flag=flag,
            warehouse_id=warehouse_id)

        product = self.env['product.product'].browse(product_id)
        warehouse = self.env['stock.warehouse'].browse(warehouse_id)
        location = self._select_lot_stock_location(product, warehouse)
        # DO NOT use 'product_id' AFTER THIS LIMIT ; only use 'product'
        product = self._select_lot_product(product)

        # Search all lot existing lot for the product and location selected
        quants = self.env['stock.quant'].read_group([
            ('product_id', '=', product.id),
            ('location_id', 'child_of', location.id),
            ('qty', '>', 0),
            ('lot_id', '!=', False),
            ], ['lot_id'], 'lot_id')
        available_lots = [quant['lot_id'][0] for quant in quants]
        res.setdefault('domain', {}).update(
            {'lot_id': [('id', 'in', available_lots)]})
        res['value']['lot_id'] = False
        return res
```

jabibi commented:
I have some minor fixes / improvements that if you want I can push to your branch

alexis-via commented:
@jabibi Yes, don't hesitate to make a PR to my branch.

jabibi committed:
[IMP] Display selection label instead of key, for rental display name

jabibi committed:
[FIX] Complete @api.constrains dependencies for rented products

jabibi committed:
[I18N] spanish translation

jabibi commented:
PR akretion/sale-workflow#4

I did not say, but you do a great job @alexis-via

Commit:
Merge pull request #4 from jabibi/8.0-add-sale_rental_lot_selection

bug fixes, minor improvements and spanish translation

pedrobaeza commented:
Please fix conflicts

rousseldenis commented:
Closing this as no activity. @alexis-via Please feel free to reopen it if you still need it. Take care of your PR's
